File: MeloRenta/src/paquete2/Inquilino.java
package paquete2;

import java.util.Date;

public class Inquilino {
    private String documento_identidad;
    private String nombre_completo;
    private String genero;
    private Date fecha_nacimiento;
    private String telefono;
    private String correo;
    private Contrato contrato;

    public Inquilino(String documento_identidad, String nombre_completo,
                     String genero, Date fecha_nacimiento, String telefono,
                     String correo, Contrato contrato) {

        this.documento_identidad = documento_identidad;
        this.nombre_completo = nombre_completo;
        this.genero = genero;
        this.fecha_nacimiento = fecha_nacimiento;
        this.telefono = telefono;
        this.correo = correo;
        this.contrato = contrato;
    }

    //getter and setter

    public String getDocumento_identidad() {
        return documento_identidad;
    }

    public void setDocumento_identidad(String documento_identidad) {
        this.documento_identidad = documento_identidad;
    }

    public String getNombre_completo() {
        return nombre_completo;
    }

    public void setNombre_completo(String nombre_completo) {
        this.nombre_completo = nombre_completo;
    }

    public String getGenero() {
        return genero;
    }

    public void setGenero(String genero) {
        this.genero = genero;
    }

    public Date getFecha_nacimiento() {
        return fecha_nacimiento;
    }

    public void setFecha_nacimiento(Date fecha_nacimiento) {
        this.fecha_nacimiento = fecha_nacimiento;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public Contrato getContrato() {
        return contrato;
    }

    public void setContrato(Contrato contrato) {
        this.contrato = contrato;
    }

    //fin getter and setter

    
    //falta el metodo toString
    @Override
    public String toString() {
        return "Inquilino{" +
                "documento_identidad=" + documento_identidad +
                ", nombre_completo=" + nombre_completo +
                ", genero=" + genero +
                ", fecha_nacimiento=" + fecha_nacimiento +
                ", telefono=" + telefono +
                ", correo=" + correo +
                ", contrato=" + contrato +
                '}';
    }
}

File: MeloRentaPython/gestorAplicacion/gestorAplicacion_contratos/Inquilino.py
import datetime as dt

class Inquilino():
    lista_inquilinos=[]
    def __init__(self, documentoIdentidad, nombreCompleto, genero, fechaNacimiento, telefono, correo, contrato):
        self._documentoIdentidad = documentoIdentidad
        self._nombreCompleto = nombreCompleto
        self._genero = genero
        self._fechaNacimiento = dt.datetime.strptime(fechaNacimiento, "%d/%m/%y")
        self._telefono = telefono
        self._correo = correo
        self._contrato = contrato
        Inquilino.lista_inquilinos.append(self)

    # Getters and Setters

    def getDocumentoIdentidad(self):
        return self._documentoIdentidad

    def setDocumentoIdentidad(self, documentoIdentidad):
        self._documentoIdentidad = documentoIdentidad

    def getNombreCompleto(self):
        return self._nombreCompleto

    def setNombreCompleto(self, nombreCompleto):
        self._nombreCompleto = nombreCompleto

    def getGenero(self):
        return self._genero

    def setGenero(self, genero):
        self._genero = genero

    def getFechaNacimiento(self):
        return self._fechaNacimiento

    def setFechaNacimiento(self, fechaNacimiento):
        self._fechaNacimiento = dt.datetime.strptime(fechaNacimiento, "%d/%m/%y")

    def getTelefono(self):
        return self._telefono

    def setTelefono(self, telefono):
        self._telefono = telefono

    def getCorreo(self):
        return self._correo

    def setCorreo(self, correo):
        self._correo = correo

    def getContrato(self):
        return self._contrato

    def setContrato(self, contrato):
        self._contrato = contrato

    # Fin Getters and Setters

    def __str__(self):
        return "Inquilino {documento identidad = " + self.getDocumentoIdentidad() + ", nombre completo = " + self.getNombreCompleto() + ", genero = " + self.getGenero() + "\n, fecha nacimiento = " + str(self.getFechaNacimiento()) + ", telefono = " + self.getTelefono() + ", correo" + self.getCorreo() + "\n, contrato = " + self.getContrato().mostrarContrato()